### ISSUE TITLE
projectile should not add my home directory to known project list

### ISSUE THREAD

**Issue by dackon** [bbatsov/projectile#1577]
Projectile added my home directory to the project list, I checked the [file markers](https://docs.projectile.mx/projectile/projects.html), not one in my home directory. But I have a folder 'Workspace' in my home directory, M-x projectile-project-info shows:

```terminal
Project dir: /Users/foobar/ ## Project VCS: none ## Project type: bazel
```

So I customized the variable and delete the WORKSPACE from the variable:
```lisp
(custom-set-variables
  '(projectile-project-root-files
    '("pubspec.yaml" "info.rkt" "Cargo.toml" "stack.yaml" "DESCRIPTION" "Eldev" "Cask" "shard.yml" "Gemfile" ".bloop" "deps.edn" "build.boot" "project.clj" "build.sbt" "application.properties" "gradlew" "build.gradle" "pom.xml" "poetry.lock" "Pipfile" "tox.ini" "setup.py" "requirements.txt" "manage.py" "angular.json" "package.json" "gulpfile.js" "Gruntfile.js" "mix.exs" "rebar.config" "composer.json" "CMakeLists.txt" "Makefile" "default.nix" "meson.build" "SConstruct" "GTAGS" "TAGS" "configure.ac" "configure.in" "cscope.out")))
```
 
## Expected behavior
Now, projectile should not recognize my home folder as a bazel project

## Actual behavior
projectile still recognize my home folder as a bazel project

### Projectile version information
```
Projectile 20200822.822
```

### Emacs version
GNU Emacs 27.1 (build 1, x86_64-apple-darwin18.7.0, NS appkit-1671.60 Version 10.14.6 (Build 18G95)) of 2020-08-12

### Operating system

macOS 10.15.6

**Comment by dackon** [comment]
I deleted these lines from projectile.el, and projectile works as expected.

```lisp
(projectile-register-project-type 'bazel '("WORKSPACE")
                                  :project-file "WORKSPACE"
                                  :compile "bazel build"
                                  :test "bazel test"
                                  :run "bazel run")
```

**Comment by bbatsov** [comment]
I'll close this ticket, as there's the workaround to you mentioned, plus one can just adjust `projectile-project-root-files` and remove `WORKSPACE` from it. As no one else reported such issue, I'll assume it's rather uncommon. If only they chose a better name for those `bazel` project descriptors...

**Comment by Be-ngt-oH** [comment]
I think `~/workspace/` used to be the default directory used by Eclipse (at least this is where I got my habit from). One way to fix this would be to use a function for bazel that checks for the existence of the file and that it's not a directory. However, this doesn't fix other incarnations of the same issue. The other thing I've noticed that it's only a problem on MacOS where the filesystem is case insensitive (that is unless you've actually got a WORKSPACE directory).

I added the following to my config which is a workable solution for me:
```elisp
(if (string-equal system-type "darwin") (delete "WORKSPACE" projectile-project-root-files))
```
